File: boj/python/daily/251217.py
# 1764 듣보잡
import sys
sys.stdin = open("input.txt", "r")

N,M = [int(x) for x in sys.stdin.readline().split(' ')]
input = [x.rstrip() for x in sys.stdin.readlines()]   

setN= set()
setM= set()

for index, value in enumerate(input):
    if((int(index))<N):
        setN.add(value)
    else:
        setM.add(value)

answer = setN & setM

result = sorted(answer)

print(len(result))
print('\n'.join(result))

File: boj/js/daily/251217.js
//1764 듣보잡
const fs = require('fs')
const path = require('path')
const filePath = path.join(__dirname, '../input.txt')
const input = fs.readFileSync(filePath).toString().trim().split('\r\n')

function A_1764() {
   const [N, M] = input[0].split(' ').map(Number)

   const setN = new Set()
   const setM = new Set()
   for (let i = 1; i <= N; i++) {
      setN.add(input[i])
   }
   for (let i = N + 1; i < input.length; i++) {
      setM.add(input[i])
   }
   const answer = []
   for (const name of setN) {
      if (setM.has(name)) answer.push(name)
   }

   // const answer = [...setN.intersection(setM)]

   console.log(answer.length + '\n' + answer.sort().join('\n'))
}
